### PR TITLE
Skip precompilation for the visual with ES6 modules

### PR DESCRIPTION
**Pull request by zBritva**

### CODE REVIEW

File: bin/pbiviz-start.js [hunk 1]
```diff
@@ -52,6 +56,13 @@ program.parse(options);
 let cwd = process.cwd();
 let server, builder;
 
+if (!CertificateTools.getCertFile(config, true)) {
+    CertificateTools.createCertFile(config);
```
Comment (ignatvilesov) [inline bin/pbiviz-start.js]: Will it ask developer to install a cert?

File: lib/CertificateTools.js [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        ConsoleWriter.error('Unknown platform. Please place a custom-generated certificate in:', certPath);
+    }
+}
+
+function getCertFile(config, silent) {
+    if (typeof silent === "undefined") {
+        silent = false;
+    }
+    let  cert = path.join(__dirname, '..', config.server.certificate);
+    let  pfx = path.join(__dirname, '..', config.server.pfx);
```
Comment (ignatvilesov) [inline lib/CertificateTools.js]: There're two spaces instead of single one

File: lib/webpack.config.js [hunk 3]
```diff
@@ -35,6 +46,8 @@ module.exports = {
         }
     },
     plugins: [
-
+        new webpack.ProvidePlugin({
+            'Globalize': "globalize/lib/globalize"
```
Comment (ignatvilesov) [inline lib/webpack.config.js]: What will happen if Globalize is not installed in node modules? Will it fail compilation?

File: bin/pbiviz-start.js (hunk 1 above)
Comment (zBritva) [reply bin/pbiviz-start.js]: No

File: lib/webpack.config.js (hunk 3 above)
Comment (zBritva) [reply lib/webpack.config.js]: No, "globalize/lib/globalize" should be imported in code

File: lib/WebPackWrap.js [hunk 4]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
 
             WebPackGenerator.prepareFolders(visualPackage);
             
-            TypescriptCompiler
+            // new style
+            if (tsconfig.compilerOptions.outDir) {
+                LessCompiler.build(visualPackage, options)
+                .then(() => WebPackGenerator.prepareWebPackConfig(visualPackage, options, tsconfig))
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

File: lib/WebPackWrap.js [hunk 5]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
+                .then(
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

Comment (ignatvilesov) [reply lib/WebPackWrap.js]: You can simplify this by
```javascript
.then(resolve)
```

File: lib/WebPackWrap.js [hunk 6]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
+                    (webpackConfig) => resolve(webpackConfig)
+                );
+                // old style
+            } else {
+                TypescriptCompiler
                 .runWatcher(tsconfig.files, tsconfig.compilerOptions, !options.devMode)
                 .then(() => LessCompiler.build(visualPackage, options))
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

File: lib/WebPackWrap.js [hunk 7]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
                 .then(() => WebPackGenerator.appendExportPowerBINameSpace(visualPackage, tsconfig))
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

File: lib/WebPackWrap.js [hunk 8]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
                 .then(() => WebPackGenerator.prepareWebPackConfig(visualPackage, options, tsconfig))
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

File: lib/WebPackWrap.js [hunk 9]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
                 .then(
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: Indent must be 4 spaces

File: lib/WebPackWrap.js [hunk 10]
```diff
@@ -158,14 +153,24 @@ class WebPackGenerator {
...
                     (webpackConfig) => resolve(webpackConfig)
```
Comment (ignatvilesov) [inline lib/WebPackWrap.js]: You can simplify this by:
```javascript
.then(resolve);
```

**Pull request merged**